shed/event_streams.py

```diff
@@ -1335,28 +1347,24 @@ def _extract_info(self, docs):
             if len(self.output_info) != len(self.vals):
                 raise RuntimeError('The output_info does not match the values')
 
-    def dispatch(self, nds):
-        """Dispatch to methods expecting particular doc types.
-
-        Parameters
-        ----------
-        nds: tuple
-            Name document pair
-
-        Returns
-        -------
-        tuple:
-            New name document pair
-        """
-        name, docs = self.curate_streams(nds, False)
+    def update(self, x, who=None):
+        name, docs = self.curate_streams(x, False)
         if name == 'start':
             self.vals = list()
+            self.unseen_docs = ['start', 'descriptor', 'event', 'stop']
             self.emit_event = False
             self.keys = self.init_keys
             self.output_info = self.init_output_info
-        if name == self.document:
-            self._extract_info(docs)
-        return getattr(self, name)(docs)
+        if self.unseen_docs:
```

Comment:
This allows us to emit an event even if an event doesn't come down (helpful when pulling metadata out of things that might not have events, looking at you `Query`)